test/attr/attr_override.swift

```diff
@@ -505,3 +505,53 @@ class Index<F, T> {
 class CollectionIndex<C : Collection> : Index<C, C.I> {
   override func map(_ f: C) -> C.I {}
 }
+
+// Overrides with different generic signature
+
+protocol SR_4206_P0 {}
+
+protocol SR_4206_P1: SR_4206_P0 {}
+protocol SR_4206_P2: SR_4206_P0 {}
+
+class SR_4206_C1 {
+  func foo<T: SR_4206_P1>(param: T) {} // expected-note {{overridden declaration is here}}
+}
+
+class SR_4206_C2: SR_4206_C1 {
+  override func foo<T: SR_4206_P2>(param: T) {} // expected-error {{cannot override method 'foo' with a different generic signature}}
+}
+
+class SR_4206_C3 {}
+class SR_4206_C4<T>: SR_4206_C3 {}
+
+class SR_4206_C5 {
+  func test<E: SR_4206_C3>(_: E) {} // expected-note {{overridden declaration is here}}
+}
+class SR_4206_C6<T>: SR_4206_C5 {
+  override func test<E>(_: E) where E: SR_4206_C4<T> {} // expected-error {{cannot override method 'test' with a different generic signature}}
+}
+
+protocol SR_4206_Proto {}
+
+class GenericClass_SR_4206<T> {
+  func foo<T: SR_4206_Proto>(arg: T) {} // expected-note {{overridden declaration is here}}
+}
+
+class ConcreteClass_SR_4206: GenericClass_SR_4206<SR_4206_Proto> {
+  override func foo<T>(arg: T) {} // expected-error {{cannot override method 'foo' with a different generic signature}}
+}
+
+protocol SR_4206_Key {}
+
+protocol SR_4206_Container {
+  associatedtype Key: SR_4206_Key
+}
+
+class SR_4206_Base<Key: SR_4206_Key> {
+  func foo(forKey key: Key) throws {}
+}
+
+class SR_4206_Derived<C: SR_4206_Container> : SR_4206_Base<C.Key> {
+  typealias Key = C.Key
+  override func foo(forKey key: Key) throws {} // Ok
```

Comment:
Can you add some more tests? We should cover all combinations of:

- Base class is generic/concrete
- Subclass is generic/concrete
- Method is generic/concrete
- Override is valid/invalid

Reply:
Sure, I'll add more tests once I can get the whole thing to build and pass existing tests 😅

Reply:
Sure, but keep in mind that adding more precise tests might make it more apparent what the existing problems in the code are, too